scripts/apps/fields/custom-fields-provider.ts

```diff
@@ -18,16 +18,8 @@ interface IFieldOptions {
     previewComponent: React.ComponentClass<IPreviewComponentProps>;
 }
 
-export class CustomFields {
-    fields: {[key: string]: IFieldOptions};
+export const fields: {[key: string]: IFieldOptions} = {};
 
-    constructor() {
-        this.fields = {};
-    }
-
-    add(fieldId: string, options: IFieldOptions) {
-        this.fields[fieldId] = options;
-    }
-}
-
-export const fields = new CustomFields();
+export const customField = (fieldId: string, options: IFieldOptions) => {
```

Comment:
`addCustomField` sounds more intuitive. I think in most cases having a verb in a function name is desirable.